opencensus/stats/internal/measure_data_test.cc

```diff
@@ -0,0 +1,143 @@
+      static_cast<double>(std::accumulate(samples.begin(), samples.end(), 0)) /
+      samples.size();
+  double expected_sum_of_squared_deviation = 0;
+  for (const auto sample : samples) {
+    data.Add(sample);
+    expected_sum_of_squared_deviation += pow(sample - expected_mean, 2);
+  }
+
+  Distribution distribution = testing::TestUtils::MakeDistribution(&buckets);
+  data.AddToDistribution(&distribution);
+  EXPECT_EQ(distribution.count(), samples.size());
+  EXPECT_DOUBLE_EQ(distribution.mean(), expected_mean);
+  EXPECT_DOUBLE_EQ(distribution.sum_of_squared_deviation(),
+                   expected_sum_of_squared_deviation);
+  EXPECT_DOUBLE_EQ(distribution.min(),
+                   *std::min_element(samples.begin(), samples.end()));
+  EXPECT_DOUBLE_EQ(distribution.max(),
+                   *std::max_element(samples.begin(), samples.end()));
+}
+
+TEST(MeasureDataTest, BatchedAddToDistribution) {
+  // Tests that batching values in the MeasureData is equivalent to sequentially
+  // adding to the distribution.
+  BucketBoundaries buckets = BucketBoundaries::Exponential(7, 2, 2);
+  MeasureData data(absl::MakeSpan(&buckets, 1));
+  Distribution base_distribution =
+      testing::TestUtils::MakeDistribution(&buckets);
+  // Add some preexisting data to fully test the merge.
+  testing::TestUtils::AddToDistribution(&base_distribution, 20);
+  testing::TestUtils::AddToDistribution(&base_distribution, 10);
+
+  Distribution expected_distribution = base_distribution;
+
+  const double tolerance = 1.0 / 1000000000;
+  const int max = 100;
+  for (int i = 0; i <= max; ++i) {
+    data.Add(i);
+    testing::TestUtils::AddToDistribution(&expected_distribution, i);
+
+    Distribution actual_distribution = base_distribution;
+    data.AddToDistribution(&actual_distribution);
+
+    EXPECT_EQ(expected_distribution.count(), actual_distribution.count());
+    EXPECT_DOUBLE_EQ(expected_distribution.mean(), actual_distribution.mean());
+    EXPECT_NEAR(expected_distribution.sum_of_squared_deviation(),
+                actual_distribution.sum_of_squared_deviation(), tolerance);
+    EXPECT_DOUBLE_EQ(expected_distribution.min(), actual_distribution.min());
+    EXPECT_DOUBLE_EQ(expected_distribution.max(), actual_distribution.max());
+    EXPECT_THAT(
+        actual_distribution.bucket_counts(),
+        ::testing::ElementsAreArray(expected_distribution.bucket_counts()));
+  }
+}
+
+TEST(MeasureDataDeathTest, AddToDistributionWithUnknownBuckets) {
+  BucketBoundaries buckets = BucketBoundaries::Explicit({0, 10});
+  MeasureData data(absl::MakeSpan(&buckets, 1));
+  data.Add(1);
+
+  BucketBoundaries distribution_buckets = BucketBoundaries::Explicit({0});
+  Distribution distribution =
+      testing::TestUtils::MakeDistribution(&distribution_buckets);
+  EXPECT_DEBUG_DEATH(
+      {
+        data.AddToDistribution(&distribution);
+        EXPECT_THAT(distribution.bucket_counts(), ::testing::ElementsAre(1, 0));
+      },
+      "");
```

Comment:
If you ABSL_ASSERT(false && "Reason.") can you EXPECT("Reason")?

Reply:
Yes. It even picks up the log on the previous line, so I do not need to duplicate the error message in the log and the assert.